common/models/model-definition.js

```diff
@@ -174,6 +183,8 @@ function ready(ModelDefinition) {
       var data = Entity.getConfigFromData(relatedData[ix]);
       delete data[relation.foreignKey];
       delete data[relation.embed.key];
+      delete data[relation.embed.keySetter];
+      delete data[relation.embed.keyGetter];
```

Comment:
`keySetter` and `keyGetter` are names of functions defined on entity constructor, they shouldn't be stored in instance data. I think these two lines are redundant and should be removed.

I think once you remove these two lines, the `name` property won't be deleted from the config data. In my opinion, the `name` property should be deleted by `ModelMethod.getConfigFromData`. That way the entity (ModelMethod for example) has more control over what's being stored in the config and it should make the code easier to understand to readers that are not entirely familiar with `formatRelatedData`/